### PR TITLE
Standardize hipSOLVER documentation builds

### PR DESCRIPTION
**Pull request by samjwu**
JIRA: https://ontrack-internal.amd.com/browse/SWDEV-376997

Summary of proposed changes:

- Updates the readthedocs documentation to use the rocm-docs-core template

Rationale:

- https://github.com/RadeonOpenCompute/rocm-docs-core#purpose

Successful Build:

- https://readthedocs.com/projects/advanced-micro-devices-hipsolver/builds/1353091/

### CODE REVIEW

File: docs/data/rocmlogo.png
Comment (saadrahim) [inline docs/data/rocmlogo.png:1]: I believe you can delete this file

**Review by saadrahim — CHANGES_REQUESTED** [review]
Minor changes to tidy up.

File: .gitignore [hunk 1]
```diff
@@ -31,11 +31,15 @@
 # Editors
 .vscode
 
-# build-in-source directory
-build/
```
Comment (cgmb) [inline .gitignore:35]: This is used for the library build.

Comment (saadrahim) [reply .gitignore:35]: FYI, @samjwu we plan to move the documentation builds to a subfolder here in the future.

File: docs/data/rocmlogo.png
Comment (samjwu) [reply docs/data/rocmlogo.png:1]: Done 
[cleanup and fixes](https://github.com/ROCmSoftwarePlatform/hipSOLVER/pull/149/commits/5420c3d914738979c2dc5c97b6de7d1a317efe34)

File: docs/index.rst
[the remark this answers is not in the document]
Comment (samjwu) [reply docs/index.rst:1]: Done 
[cleanup and fixes](https://github.com/ROCmSoftwarePlatform/hipSOLVER/pull/149/commits/5420c3d914738979c2dc5c97b6de7d1a317efe34)

File: .gitignore (hunk 1 above)
Comment (samjwu) [reply .gitignore]: I forgot about this; added it back in
[cleanup and fixes](https://github.com/ROCmSoftwarePlatform/hipSOLVER/pull/149/commits/5420c3d914738979c2dc5c97b6de7d1a317efe34)

**Review by saadrahim — APPROVED** [review]

**Review by cgmb — APPROVED** [review]

**Pull request merged**